src/main/ngapp/src/app/dataservices/connection-node-selector/connection-node-selector.component.ts

```diff
@@ -0,0 +1,184 @@
+            const connStatus = connectionSummary.getStatus();
+            const conn = connectionSummary.getConnection();
+            conn.setStatus(connStatus);
+            conns.push(conn);
+          }
+          self.connections = conns;
+          // Set the root nodes of the tree (Connections)
+          const treeNodes = [];
+          for (const conn of conns) {
+            if (conn.isActive) {
+              const node = new SchemaNode();
+              node.setName(conn.getId());
+              node.setType(ConnectionsConstants.schemaNodeType_connection);
+              node.setHasChildren(true);
+              treeNodes.push(node);
+            }
+          }
+          self.connectionSchemaTree.setTreeRoots(treeNodes);
+          self.connectionLoadingState = LoadingState.LOADED_VALID;
+          if (self.wizardService.isEdit()) {
+            self.initEdit();
+          }
+        },
+        (error) => {
+          self.logger.error("[ConnectionSchemaTreeComponent] Error getting connections: %o", error);
+          self.connectionLoadingState = LoadingState.LOADED_INVALID;
+        }
+      );
+  }
+
+  /**
+   * selector is valid if at least one node is selected.
+   * @returns {boolean} the selector status (true if one or more nodes selected)
+   */
+  public valid( ): boolean {
+    return this.getSelectedNodes().length > 0;
+  }
+
+  /*
+   * Return all currently selected Nodes
+   * @returns {SchemaNode[]} the list of selected connection Nodes
+   */
+  public getSelectedNodes(): SchemaNode[] {
+    return this.wizardService.getSelectedSchemaNodes();
+  }
+
+  public onTreeNodeSelected( $event: SchemaNode ): void {
+    this.selectedTreeNode = $event;
+  }
+
+  public onTreeNodeDeselected( $event: SchemaNode ): void {
+    this.selectedTreeNode = null;
+  }
+
+  public onAddSourceNodeClicked( ): void {
+    this.wizardService.addToSelectedSchemaNodes(this.selectedTreeNode);
+    this.selectedNodesList.setNodes(this.wizardService.getSelectedSchemaNodes());
+    this.selectedNodeListUpdated.emit();
+  }
+
+  public onSourceNodeRemoved( node: SchemaNode ): void {
+    this.wizardService.removeFromSelectedSchemaNodes(node);
+    this.selectedNodesList.setNodes(this.wizardService.getSelectedSchemaNodes());
+    this.selectedNodeListUpdated.emit();
+  }
+
+  public get enableAddArrow(): boolean {
+    // Enable the add arrow if (1) a queryable node is selected and (2) it's not already in the selected list
+    return this.selectedNodeIsQueryable && !this.selectedNodesList.hasNode(this.selectedTreeNode);
+  }
+
+  public getConnectionsForSelectedNodes(): Connection[] {
+    // Array of unique connection names from the selections
+    const uniqueConnNames: string[] = [];
+    for (const node of this.getSelectedNodes()) {
+      const connName = node.getConnectionName();
+      if (uniqueConnNames.indexOf(connName) === -1) {
+        uniqueConnNames.push(connName);
+      }
```

Comment:
Maybe move code from below to here that adds the connection to "resultConns". This would avoid executing one loop. wdyt?